pyccl/p2d.py

```diff
@@ -0,0 +1,144 @@
+from . import ccllib as lib
+
+from .core import check
+import numpy as np
+
+# TODO choices about interpolation/extrapolation
+
+
+class Pk2D(object):
+    """A power spectrum class holding the information needed to reconstruct an
+    arbitrary function of wavenumber and scale factor.
+    """
+    def __init__(self, pkfunc=None, a_arr=None, lk_arr=None, pk_arr=None,
+                 is_logp=True, interp_order_lok=1, interp_order_hik=2):
+        """Constructor for Pk2D objects.
```

Comment:
Doc strings for the init go in the class level doc string, not on the init.

Reply:
Fixed